pipeline/resolve.go

```diff
@@ -122,6 +122,10 @@ func resolveStartBlockNum(ctx context.Context, req *pbsubstreamsrpc.Request, res
 		return 0, "", nil, status.Errorf(grpccodes.InvalidArgument, "invalid StartCursor %q: %s", cursor, err.Error())
 	}
 
+	if req.StopBlockNum > 0 && req.StopBlockNum < cursor.Block.Num() {
+		return 0, "", nil, status.Errorf(grpccodes.InvalidArgument, "StartCursor %q is after StopBlockNum %d", cursor, req.StopBlockNum)
+	}
+
```

Comment:
Just to be sure, did you validate this correctly yields `grpc.InvalidArgument` from consumer perspective at the `gRPC` level?

Reply:
```InvalidArgument desc = StartCursor "c1:1:18042010:c3a51e1c6e24acda78dedf60830aa1bc9fa76de141fa0304888e3de853940a69:18041945:a01338b59ae0b0f9c0e774bb64827d949455b6909f4fcfc69116d785cb197fb4" is after StopBlockNum 18041955```

yes

Reply:
Thanks!